src/v2/Apps/Order/Components/SavedAddresses.tsx

```diff
@@ -89,15 +38,99 @@ const defaultAddressIndex = addressList => {
   return `${indexOfDefaultAddress > -1 ? indexOfDefaultAddress : 0}`
 }
 
-interface Props {
-  me: SavedAddresses_me
-  onSelect: (string) => void
-  handleClickEdit: (number) => void
-}
+const SavedAddresses: React.FC<SavedAddressesProps> = props => {
+  const [showAddressModal, setShowAddressModal] = useState(false)
+  const logger = createLogger("SavedAddresses.tsx")
+
+  const { onSelect, handleClickEdit, me, inCollectorProfile, relay } = props
+  const addressList = me?.addressConnection?.edges ?? []
+  const handleModal = () => {
+    setShowAddressModal(!showAddressModal)
+  }
+  const collectorProfileAddressItems = addressList.map((address, index) => {
+    if (!address.node) {
+      return null
+    }
 
-const SavedAddresses: React.FC<Props> = props => {
-  const { onSelect, handleClickEdit, me } = props
-  const addressList = me.addressConnection.edges
+    const isDefaultAddress = address.node.isDefault
+
+    return (
+      <BorderBox
+        p={2}
+        width="100%"
+        flexDirection="column"
+        key={"addressIndex" + index}
+      >
+        <SavedAddressItem
+          index={index}
+          address={address.node}
+          handleClickEdit={handleClickEdit}
+        />
+        <Separator my={1} />
+        <ModifyAddressWrapper>
+          {!isDefaultAddress && (
+            <Box mr={1}>
+              <Text variant="text" color="black60">
+                Set as Default
+              </Text>
+            </Box>
+          )}
+          <Box mr={1}>
+            <Text variant="text" color="blue100">
+              Edit
+            </Text>
+          </Box>
+          <Box>
+            <Text variant="text" color="red100">
+              Delete
+            </Text>
+          </Box>
+        </ModifyAddressWrapper>
+      </BorderBox>
+    )
+  })
+
+  const addAddressButton = (
+    <>
+      <Button
+        mt={addressList.length > 0 ? 3 : 0}
+        variant="primaryBlack"
+        size="large"
+        onClick={() => handleModal()}
+      >
+        Add new address
+      </Button>
+      {showAddressModal && (
+        <AddressModal
```

Comment:
#minor it would have been great not to have the `<AddressModal />` included in two different places. It comes from `shipping/index.tsx` for checkout but here for collector profile.